### PR TITLE
Beamprocess manual override

### PR DESCRIPTION
**Pull request by JoschD**
Tested (with Mael I think?) in the CCC when the wrong BP was found.
I also fixed some deprecated use of `get_loc` in `pandas 2.0` and changed `append()`s
 to `concat()`s.

Most of the files changed are:
 - documentation
 - github workflows
 - remove hdf5
 
 In an ideal world I would have merged these beforehand... Should be quick to check.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -46,4 +39,4 @@ jobs:
         run: python -m pip install '.[test]'
 
       - name: Run basic tests
-        run: python -m pytest -m "not cern_network" --mpl --cov-report xml --cov=pylhc
+        run: python -m pytest -m "not cern_network" --mpl
```
Comment (fsoubelet) [inline .github/workflows/tests.yml:42]: Why the removal, are these default options now?

File: pylhc/machine_settings_info.py [hunk 2]
```diff
@@ -94,13 +98,18 @@ def _get_params() -> dict:
             help="List of knobnames. "
                  "If `None` (or omitted) no knobs will be extracted. "
                  "If it is just the string ``'all'``, "
-                 "all knobs will be extracted (can be slow)."
+                 "all knobs will be extracted (can be slow). "
                  "Use the string ``'default'`` the main knobs of interest."
```
Comment (fsoubelet) [inline pylhc/machine_settings_info.py:102]: Feels like there's a missing word in here. Maybe "Use the string ``'default'`` for the main pre-defined knobs of interest."

File: pylhc/machine_settings_info.py [hunk 3]
```diff
@@ -125,71 +134,79 @@ def get_info(opt) -> Dict[str, object]:
...
-            If it is just the string ``'all'``,
-            all knobs will be extracted (can be slow).
-            Use the string ``'default'`` the main knobs of interest.
-            If this is called from python, the strings need
-            to be put as single items into a list.
+        List of knobnames.
+        If `None` (or omitted) no knobs will be extracted.
+        If it is just the string ``'all'``,
+        all knobs will be extracted (can be slow).
+        Use the string ``'default'`` the main knobs of interest.
```
Comment (fsoubelet) [inline pylhc/machine_settings_info.py:167]: Same here

File: pylhc/machine_settings_info.py [hunk 4]
```diff
@@ -394,19 +416,24 @@ def write_trim_histories(
...
+    bp_info.Fill = fill_no
+
     try:
-        start_time = _get_beamprocess_start(fill_bps, acc_time, str(beamprocess))
+        start_time = _get_beamprocess_start(fill_bps, acc_time, bp_info.Name)
     except ValueError as e:
         raise ValueError(f"In fill {fill_no} the {str(e)}") from e
-    bp_info = LSA.get_beamprocess_info(beamprocess)
-    bp_info.update({"Fill": fill_no, "StartTime": start_time})
+    bp_info.StartTime = start_time
```
Comment (fsoubelet) [inline pylhc/machine_settings_info.py:433]: Should this line be in the `try` block? It seems to me like `AccDatetime` initiation will not return a value if an error occurs.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (JoschD) [reply .github/workflows/tests.yml:42]: why were they there in the first place?

File: pylhc/machine_settings_info.py (hunk 4 above)
Comment (JoschD) [reply pylhc/machine_settings_info.py:433]: No, the error is raised anyway. Only excepted to add info

File: .github/workflows/tests.yml (hunk 1 above)
Comment (fsoubelet) [reply .github/workflows/tests.yml:42]: Aah this is not the coverage workflow, my bad! All good

File: pylhc/machine_settings_info.py [hunk 5]
```diff
@@ -17,28 +17,32 @@
...
+  -h, --help            show this help message and exit
+  --time TIME           UTC Time as 'Y-m-d H:M:S.f' or ISO format or AccDatetime object.
+                        Acts as point in time or end time (if ``start_time`` is given).
+  --start_time START_TIME
+                        UTC Time as 'Y-m-d H:M:S.f' or ISO format or AccDatetime object.
+                        Defines the beginning of the time-range.
+  --knobs KNOBS [KNOBS ...]
+                        List of knobnames. If `None` (or omitted) no knobs will be extracted.
+                        If it is just the string ``'all'``, all knobs will be extracted
+                        (can be slow). Use the string ``'default'`` the main knobs
```
Comment (fsoubelet) [inline pylhc/machine_settings_info.py:35]: Not updated here, but that's ok

**Review by fsoubelet — APPROVED** [review]

File: pylhc/machine_settings_info.py (hunk 5 above)
Comment (JoschD) [reply pylhc/machine_settings_info.py:35]: fuck. I thought I checked there and thought it used a shorter description. No idea

**Review by fsoubelet — APPROVED** [review]

**Pull request merged**